src/Microsoft.ML.TimeSeries/SequentialAnomalyDetectionTransformBase.cs

```diff
@@ -16,678 +16,676 @@
 
 namespace Microsoft.ML.TimeSeriesProcessing
 {
-    // REVIEW: This base class and its children classes generate one output column of type VBuffer<Double> to output 3 different anomaly scores as well as
-    // the alert flag. Ideally these 4 output information should be put in four seaparate columns instead of one VBuffer<> column. However, this is not currently
-    // possible due to our design restriction. This must be fixed in the next version and will potentially affect the children classes.
-
     /// <summary>
-    /// The base class for sequential anomaly detection transforms that supports the p-value as well as the martingales scores computation from the sequence of
-    /// raw anomaly scores whose calculation is specified by the children classes. This class also provides mechanism for the threshold-based alerting on
-    /// the raw anomaly score, the p-value score or the martingale score. Currently, this class supports Power and Mixture martingales.
-    /// For more details, please refer to http://arxiv.org/pdf/1204.3251.pdf
+    /// The type of the martingale.
     /// </summary>
-    /// <typeparam name="TInput">The type of the input sequence</typeparam>
-    /// <typeparam name="TState">The type of the state object for sequential anomaly detection. Must be a class inherited from AnomalyDetectionStateBase</typeparam>
-    public abstract class SequentialAnomalyDetectionTransformBase<TInput, TState> : SequentialTransformerBase<TInput, VBuffer<Double>, TState>
-        where TState : SequentialAnomalyDetectionTransformBase<TInput, TState>.AnomalyDetectionStateBase, new()
+    public enum MartingaleType : byte
     {
         /// <summary>
-        /// The type of the martingale.
+        /// (None) No martingale is used.
```

Comment:
Do you have a wiki link for this? It's hard to understand if user needs to read through the reference paper.

Reply:
I don't have a wiki link but we do plan to upload docs for time series that explain all these enums. It will be a markdown and it will be its own PR.

---
In reply to: [254484290](https://github.com/dotnet/machinelearning/pull/2344#discussion_r254484290) [](ancestors = 254484290)